src/main/java/com/atlassian/bitbucket/jenkins/internal/client/ModernBitbucketBuildStatusClientImpl.java

```diff
@@ -1,29 +1,53 @@
 package com.atlassian.bitbucket.jenkins.internal.client;
 
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
 import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import com.atlassian.bitbucket.jenkins.internal.provider.DefaultInstanceIdentityProvider;
+import com.atlassian.bitbucket.jenkins.internal.provider.InstanceIdentityProvider;
+import com.google.common.annotations.VisibleForTesting;
 import okhttp3.HttpUrl;
 
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.*;
+
 import static java.util.Objects.requireNonNull;
 import static org.apache.commons.lang3.StringUtils.stripToNull;
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class ModernBitbucketBuildStatusClientImpl implements BitbucketBuildStatusClient {
 
+    private static final String BUILD_STATUS_SIGNATURE_ALGORITHM_ID = "BBS-Signature-Algorithm";
+    private static final String BUILD_STATUS_SIGNATURE_ID = "BBS-Signature";
     private static final String BUILD_STATUS_VERSION = "1.0";
+    private static final String SIGNING_ALGORITHM = "SHA";
     private final BitbucketRequestExecutor bitbucketRequestExecutor;
-    private final String revisionSha;
+    private final InstanceIdentityProvider instanceIdentityProvider;
     private final String projectKey;
     private final String repoSlug;
+    private final String revisionSha;
```

Comment:
It _seems_ like you have auto alphabetical sort on in IDEA, which causes unnecessary formatting changes like this and especially since this is an open source repo and we can't (or won't at least) universally enforce alphabetical ordering on all contributions, this sort of changes will become noise and potentially can cause merge conflicts in the future. I had the same problem when I started working on this repo, given we use alphabetical ordering internally, but I turned it off for this repo.

Reply:
I do, and haven't gotten around to turning it off :coneofshame: